bahmni/src/main/java/org/avni_integration_service/bahmni/contract/OpenMRSFullEncounter.java

```diff
@@ -177,4 +183,15 @@ public Object getObservationValue(String conceptUuid) {
         if (obs == null) return null;
         return obs.getValue();
     }
+
+    //TODO: This should be handled in a better way by getting and validating against /bahmnicore/observations API
+    private boolean isComplexObs(Map<String, Object> observation) {
+        Object obsValue = observation.get("value");
+        return obsValue instanceof Map  && ((Map) obsValue).get("display").toString().equals("raw file");
+    }
+
+    private String getValueforComplexObs(Map<String, Object> observation){
```

Comment:
How would you set valueComplex for coded datatype?

Reply:
This case is handled for only concepts which have datatype as `Complex` in OpenMRS. For concepts with `Coded` datatype, `openmrs/ws/rest/v1/encounter` endpoint still returns the captured value in `value` field itself. Anyways added a todo to handle this better with `/bahmnicore/observations` endpoint to map based on the datatype field of the concept.